### PR TITLE
Docs module generation 2 of 3 PRs

### PR DESCRIPTION
**Pull request by blackboxsw**
Continued migration for cc_module documentation from python MetaSchema to separate rtd/module-docs/*/data.yaml files

Modules migrated:
cc_bootcmd, cc_byobu, cc_ca_certs, cc_chef, cc_disable_ec2_metadata, cc_fan, cc_final_message, cc_growpart, cc_grub_dpkg, cc_install_hotplug, cc_keyboard, cc_keys_to_console, cc_landscape, cc_locale, cc_lxd, cc_mcollective, cc_mounts, cc_ntp, cc_package_update_upgrade_install, cc_phone_home, cc_power_state_change, cc_puppet, cc_resizefs, cc_resolv_conf, cc_rh_subscription

## Additional ContextContinued migration for cc_module documentation from python MetaSchema to separate rtd/module-docs/*/data.yaml files

Modules migrated:
cc_bootcmd, cc_byobu, cc_ca_certs, cc_chef, cc_disable_ec2_metadata, cc_fan, cc_final_message, cc_growpart, cc_grub_dpkg, cc_install_hotplug, cc_keyboard, cc_keys_to_console, cc_landscape, cc_locale, cc_lxd, cc_mcollective, cc_mounts, cc_ntp, cc_package_update_upgrade_install, cc_phone_home, cc_power_state_change, cc_puppet, cc_resizefs, cc_resolv_conf, cc_rh_subscription

## Additional Context

commit 4c035f02d5a372e2e20a1592a1fe4359c3af8350 adds deprecation and changed templates
commit 7fba14017b0bcc079e88096ab64a81368efe903a makes `flatten_schema_all_of` and 
`flatten_schema_defs` public so the functions can be shared between `cloud-init schema --docs` and RTD doc generation.

Expect two more PRs after this:
- 1. which migrates the remaining cc_modules to templates
- 2. which drops unused functions from cloudinit.schema and finalized MetaSchema typing.

## Test Steps
```
rm -rf doc/rtd_html/;  tox -e docs
xdg-open doc/rtd_html/reference/modules.html
```

## Checklist

- [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/index.html)
- [ ] I have updated or added any [unit tests](https://cloudinit.readthedocs.io/en/latest/development/testing.html) accordingly
- [ ] I have updated or added any [documentation](https://cloudinit.readthedocs.io/en/latest/development/contribute_docs.html) accordingly

## Merge type

- [ ] Squash merge using "Proposed Commit Message"
- [x] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

**Review by TheRealFalcon — COMMENTED** [review]
You scared me with 449 files!

You included an `rtd_html_main` directory where you shouldn't have. Does that need to be `.gitignore`ed?

File: doc/module-docs/cc_mounts/data.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        of ``/abc/def``) then this will not work as expected - ``cc_mounts``
+        first creates the directories for all the mountpoints **before** it
+        starts to perform any mounts and so the sub-mountpoint directory will
+        not be created correctly inside the parent mountpoint.
+
+        For systems using util-linux's ``mount`` program this issue can be
+        worked around by specifying ``X-mount.mkdir`` as part of a
+        ``fs_mntops`` value for the subsequent mount entry.
+  examples:
+  - comment: |
```
Comment (TheRealFalcon) [inline doc/module-docs/cc_mounts/data.yaml:54]: I don't think we need to fix it in this PR, but the `|` causes additional newlines to get added to the output.

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Generally LGTM...just some inline typos.

Can we run a yaml formatter across the new yaml example files? Indentation isn't consistent across them.

Also, I'm looking at ways of handling example text before an example. For one-line comments in the earlier modules it looked fine, but the formatting for a longer comments don't look great to me. In particular, the text in three examples look bad to me:

cc_apt_configure
cc_lxd
cc_mounts

E.g., with apt_configure:
image

I tried manually formatting the pre-example text in `cc_apt_configure`, but if I use a yaml text block with `|`, newlines get inserted between each line, which is also not what I want. I don't think it needs to change in this PR, but it's one more thing we could improve as part of this overall effort.

File: doc/module-docs/cc_mounts/data.yaml (hunk 1 above)
Comment (blackboxsw) [reply doc/module-docs/cc_mounts/data.yaml:54]: Will take this work item to the next PR and I'll add a tool to standarize the formatting of all data.yaml and example*yaml files as part of this work.

Comment (s-makin) [reply doc/module-docs/cc_mounts/data.yaml:54]: I've been formatting the yaml examples as part of my examples PR, I noticed the same issues - but having an automatic check would be excellent

**Review by s-makin — APPROVED** [review]
Overall lgtm - I can start to see the shape this is taking, so I will start updating my PR accordingly. I think it's fine to leave the polishing/tidying as future work.

**Review by TheRealFalcon — APPROVED** [review]
LGTM (assuming lints pass)!

**Pull request merged**